Add demo solar forecast template

terrorobe opened the pull request:
I'm trying to fix a bug where the accumulated forecast energy doesn't get restored properly and eventually work on caching forecast information across (re-)starts to prevent running into API rate limits.

To better be able to work on these features I was missing a way to mock solar forecast data.

I was torn between either extending the fixed tariff implementation to also be able to emit solar forecasts or the javascript approach.

The javascript approach ended up being more composable so I went with that direction.

---

Example screenshots from the default forecast data:


Curve
![Screenshot 2025-06-21 at 19 24 41](https://github.com/user-attachments/assets/2a85d8b2-68db-4455-9001-909c9345b2c1)

Static
![Screenshot 2025-06-21 at 19 22 57](https://github.com/user-attachments/assets/216df824-f770-485d-b52a-41fe063b9bf8)

terrorobe committed:
Add solar forecasts suitable for testing

terrorobe committed:
Department of second thoughts: use 3 days

andig commented:
`demo-solar-tariff` would be a name in line with other demo- templates. From user perspective this would only apply for the bell curve tariff though? Do we need both templates or can we stick with that one? 

/cc @naltatis

terrorobe committed:
Allow for continuous sun

terrorobe committed:
Rename to demo for better legibility

terrorobe committed:
Delete no longer needed static forecast

terrorobe committed:
Update information

terrorobe committed:
More compact

terrorobe committed:
Consolidate logic

terrorobe committed:
Maybe we don't need the test after all

terrorobe committed:
The curve is probably redundant at this point

terrorobe commented:
I removed the test and the static forecast and renamed this template to just `demo-solar-forecast`


---

Here's a few examples:

Continuous sun, low point between 5 and 6 (sunrise 6, sunset 5)

![Screenshot 2025-06-23 at 22 57 33](https://github.com/user-attachments/assets/6c435553-4582-4cfe-964d-5ba7cc978083)

Spanning midnight, (sunrise 22, sunset 5)
![Screenshot 2025-06-23 at 22 58 20](https://github.com/user-attachments/assets/fdc0917c-050e-4e03-a7b2-2de7d9876f01)


The sun is gone, (any sunrise === sunset combination)
![Screenshot 2025-06-23 at 22 57 48](https://github.com/user-attachments/assets/a4e36507-aa32-480a-b8f7-c9cd4d473d06)

terrorobe committed:
Use standardized unit name, fix template name

naltatis committed:
Merge branch 'master' into terrorobe/solar-forecast-tests

naltatis committed:
align property descriptioins; add to demo mode

andig commented:
Danke @naltatis für den Einbau in den Demomodus. Da freut sich auch Claude ;)

naltatis commented:
Nice PR! Thanks @terrorobe 🙌.

I've made a few changes and added it to our demo config.

andig commented:
@terrorobe das wäre auch ein interessanter Feed-in Tarif zu Demozwecken wenn man ihn mit Offset oder Vorzeichen versorgen könnte (=Mittags kostet einspeisen).

naltatis committed:
simplify

naltatis commented:
Find ich ne gute Idee.
